arangod/Aql/RestAqlHandler.h

```diff
@@ -136,6 +136,16 @@ class RestAqlHandler : public RestVocbaseBaseHandler {
   // our query registry
   QueryRegistry* _queryRegistry;
 
+  // Guard that keeps the lock on the QueryContext to serialize
+  // concurrent snippet execution
+  struct QueryGuard {
```

Comment:
Probably it would be better to move the `QueryGuard` into QueryContext.h, and probably even allow creating one by calling a public method on the `QueryContext` class. Then we don't need to publicly expose `lock()` and `unlock()` methods in QueryContext at all, and the only way to get a lock would be by calling the one public method that returns a RAII `QueryGuard` object.